crates/runtime/src/externref.rs

```diff
@@ -797,19 +798,36 @@ impl StackMapRegistry {
         let mut min = usize::max_value();
         let mut max = 0;
         let mut pc_to_stack_map = vec![];
+        let mut last_is_none_marker = true;
 
         for (range, infos) in stack_maps {
             let len = range.end - range.start;
 
             min = std::cmp::min(min, range.start);
             max = std::cmp::max(max, range.end);
 
+            // Add a marker between functions indicating that this function's pc
+            // starts with no stack map so when our binary search later on finds
+            // a pc between the start of the function and the function's first
+            // stack map it doesn't think the previous stack map is our stack
+            // map.
+            //
+            // We skip this if the previous entry pushed was also a `None`
+            // marker, in which case the starting pc already has no stack map.
+            // This is also skipped if the first `code_offset` is zero since
+            // what we'll push applies for the first pc anyway.
+            if !last_is_none_marker && (infos.is_empty() || infos[0].code_offset > 0) {
```

Comment:
Could make this a tiny bit less wordy (although perhaps some would argue less readable) like this:

```suggestion
            if !last_is_none_marker && (infos.get(0).map_or(true, |i| i.code_offset > 0) {
```

I would probably use `map_or` personally but it doesn't really matter either way.

Reply:
Oh I guess this isn't actually even less wordy... nevermind!